src/ADAL.PCL/AdalHttpClient.cs

```diff
@@ -103,6 +110,22 @@ public AdalHttpClient(string uri, CallState callState)
                         string[] errorCodes = tokenResponse.ErrorCodes ?? new[] {ex.WebResponse.StatusCode.ToString()};
                         serviceEx = new AdalServiceException(tokenResponse.Error, tokenResponse.ErrorDescription,
                             errorCodes, ex);
+
+                        if (((int) ex.WebResponse.StatusCode).Equals(500) ||
```

Comment:
Yes we can.But last time during jason's code review , it was discussed that numbers are more representative than strings for http status codes.Hence used the numbers.